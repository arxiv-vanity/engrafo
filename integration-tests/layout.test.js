var utils = require("./utils");

test("authors render correctly", done => {
  utils.expectBodyToMatchSnapshot("layout/authors.tex", done);
});

test("paragraphs and line breaks render correctly", done => {
  utils.expectBodyToMatchSnapshot("layout/paragraphs.tex", done);
});

test("footnotes render correctly", done => {
  utils.expectBodyToMatchSnapshot("layout/footnotes.tex", done);
});

test("footnotes in author render correctly", done => {
  utils.expectBodyToMatchSnapshot("layout/footnotes-author.tex", done);
})

test("dates are not rendered", done => {
  utils.expectBodyToMatchSnapshot("layout/date.tex", done);
})

test("abstracts work", done => {
  utils.expectBodyToMatchSnapshot("layout/abstract.tex", done);
});
