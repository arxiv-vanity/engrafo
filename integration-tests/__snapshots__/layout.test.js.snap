// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`abstracts work 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <h1
          class="ltx_title ltx_title_document"
        >
          Hello
        </h1>
        <div
          class="engrafo-metadata"
        >
          <div
            class="engrafo-metadata-custom"
          />
        </div>
        


        <div
          class="ltx_abstract"
        >
          

          <h2
            class=""
          >
            Abstract
          </h2>
          

          <p
            class="ltx_p"
          >
            Some abstract.
          </p>
          

          <p
            class="ltx_p"
          >
            A reference to section 
            <a
              class="ltx_ref"
              href="#S1"
              title="1 Section Title ‣ Hello"
            >
              <span
                class="ltx_text ltx_ref_tag"
              >
                1
              </span>
            </a>
            .
          </p>
          

        </div>
        

        <section
          class="ltx_section"
          id="S1"
        >
          

          <h2
            class="ltx_title ltx_title_section"
          >
            

            <span
              class="ltx_tag ltx_tag_section"
            >
              <a
                href="#S1"
              >
                1 
              </a>
            </span>
            Section Title
          </h2>
          


          <div
            class="ltx_para"
            id="S1.p1"
          >
            

            <p
              class="ltx_p"
            >
              Body text.
            </p>
            

          </div>
          

        </section>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;

exports[`authors render correctly 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <h1
          class="ltx_title ltx_title_document"
        >
          Hello
        </h1>
        <div
          class="engrafo-metadata"
        >
          <div
            class="ltx_authors"
          >
            

            <span
              class="ltx_creator ltx_role_author"
            >
              

              <span
                class="ltx_personname"
              >
                Joe Bloggs

              </span>
            </span>
            

            <span
              class="ltx_author_before"
            >
                
            </span>
            <span
              class="ltx_creator ltx_role_author"
            >
              

              <span
                class="ltx_personname"
              >
                Jane Smith

              </span>
            </span>
            

          </div>
          <div
            class="engrafo-metadata-custom"
          />
        </div>
        



        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            Body text.
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;

exports[`dates are not rendered 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <h1
          class="ltx_title ltx_title_document"
        >
          Hello
        </h1>
        <div
          class="engrafo-metadata"
        >
          <div
            class="engrafo-metadata-custom"
          />
        </div>
        



        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            Hello.
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;

exports[`footnotes in author render correctly 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <h1
          class="ltx_title ltx_title_document"
        >
          Hello
        </h1>
        <div
          class="engrafo-metadata"
        >
          <div
            class="ltx_authors"
          >
            

            <span
              class="ltx_creator ltx_role_author"
            >
              

              <span
                class="ltx_personname"
              >
                Joe Bloggs
                <dt-fn>
                  <sup>
                    <span
                      data-hover-ref="dt-fn-hover-box-0"
                      id="fn-1"
                      style="cursor:pointer"
                    >
                      1
                    </span>
                  </sup>
                </dt-fn>
                

              </span>
            </span>
            

          </div>
          <div
            class="engrafo-metadata-custom"
          />
        </div>
        



        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            Body text.
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        >
          <h3>
            Footnotes
          </h3>
          <dt-fn-list>
            <ol>
              <li>
                <span
                  class="ltx_note_outer"
                >
                  <span
                    class="ltx_note_content"
                  >
                    Equal contribution.
                  </span>
                </span>
              </li>
            </ol>
          </dt-fn-list>
        </div>
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    >
      <div
        class="dt-hover-box"
        id="dt-fn-hover-box-0"
        style="display:none;"
      >
        <span
          class="ltx_note_outer"
        >
          <span
            class="ltx_note_content"
          >
            Equal contribution.
          </span>
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`footnotes render correctly 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            This is some text.
            <dt-fn>
              <sup>
                <span
                  data-hover-ref="dt-fn-hover-box-0"
                  id="fn-1"
                  style="cursor:pointer"
                >
                  1
                </span>
              </sup>
            </dt-fn>
             This is some more text.
            <dt-fn>
              <sup>
                <span
                  data-hover-ref="dt-fn-hover-box-1"
                  id="fn-2"
                  style="cursor:pointer"
                >
                  2
                </span>
              </sup>
            </dt-fn>
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        >
          <h3>
            Footnotes
          </h3>
          <dt-fn-list>
            <ol>
              <li>
                <span
                  class="ltx_note_outer"
                >
                  <span
                    class="ltx_note_content"
                  >
                    This is a footnote.
                  </span>
                </span>
              </li>
              <li>
                <span
                  class="ltx_note_outer"
                >
                  <span
                    class="ltx_note_content"
                  >
                    And a footnote with a
second paragraph! And some 
                    <em
                      class="ltx_emph ltx_font_italic"
                    >
                      formatting
                    </em>
                    .
                  </span>
                </span>
              </li>
            </ol>
          </dt-fn-list>
        </div>
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    >
      <div
        class="dt-hover-box"
        id="dt-fn-hover-box-0"
        style="display:none;"
      >
        <span
          class="ltx_note_outer"
        >
          <span
            class="ltx_note_content"
          >
            This is a footnote.
          </span>
        </span>
      </div>
      <div
        class="dt-hover-box"
        id="dt-fn-hover-box-1"
        style="display:none;"
      >
        <span
          class="ltx_note_outer"
        >
          <span
            class="ltx_note_content"
          >
            And a footnote with a
second paragraph! And some 
            <em
              class="ltx_emph ltx_font_italic"
            >
              formatting
            </em>
            .
          </span>
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`paragraphs and line breaks render correctly 1`] = `
<body>
  <div
    class="engrafo-container"
  >
    


    <div
      class="ltx_page_content"
    >
      

      <dt-article>
        

        <div
          class="ltx_para"
          id="p1"
        >
          

          <p
            class="ltx_p"
          >
            This is the first paragraph.
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p2"
        >
          

          <p
            class="ltx_p"
          >
            This is the second paragraph.
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p3"
        >
          

          <p
            class="ltx_p ltx_align_center"
          >
            This is a centered paragraph.
          </p>
          

        </div>
        

        <div
          class="ltx_para"
          id="p4"
        >
          

          <p
            class="ltx_p"
          >
            This is a

            <br
              class="ltx_break"
            />
            line break.
          </p>
          

        </div>
        

      </dt-article>
      

      <dt-appendix>
        

        


        <div
          class="l-body"
        />
        

      </dt-appendix>
    </div>
    





    <div
      id="footnote-hover-boxes-container"
    />
  </div>
</body>
`;
